### ISSUE TITLE
Can't do Register Filtering Query through validation of filter query

### ISSUE THREAD

**Issue by Oleh-Shok** [anton-martyniuk/Modern#8]
**Conditions:**
_using Modern.Repositories.Abstractions, Version=1.0.0.0_

**Stages of reproduction:**
1. _Create specifications - need two or more._
> 
>  new FilterByClientSpecification(x.Client),
>>AddFilteringQuery(x => x.Client != null
    && clients.Contains(x.Client.Name, StringComparer.Ordinal));

>  new FilterByCustomerSpecification(x.Customer),
>> AddFilteringQuery(
     x => x.Customer != null && customers.Any(t => x.Customer.Contains(t, StringComparison.OrdinalIgnoreCase)));
3. _Register specifications into EfCoreSpecification<Database>_
>
> var registerSpecification = specifications.Aggregate((current, next) => current.And(next));
> var efCoreSpecification = new EfCoreSpecification<Database>(registerSpecification );
_4. Send a request POST with selected specifications_

**Expected result:**
_We expect filtering with selected specifications: especially if one have been chosen_

**Actual result:**
_Can't do Register Filtering Query through validation of filter query. In that method:_
 
> private void RegisterFilteringQuery(Specification<TEntity> left, Specification<TEntity> right)

**Comment by anton-martyniuk** [comment]
@Oleh-Shok I've fixed the issue, make sure to update the `Modern.Repositories.Abstractions package` to version **1.0.1** and all other packages that reference it. for a full list of updated packages see a list in a commit down below

**Comment by anton-martyniuk** [comment]
@Oleh-Shok I've reworked my packages and now main packages are as follows: Modern.Repositories.EFCore that has DI extensions.
Previous package: Modern.Repositories.EFCore.DependencyInjection is deprecated and will no longer be maintained.
This applies to all packages, all DependencyInjection are deprecated.

I've released new version of packages with version 2.0.0-beta1. I've tested them and I am looking forward if they'll work for you too. If so, I plan to release them the next week.

Make corresponding updates of packages in your projects.

Commits:
c8617ddddd63e3785ece97e7fcf417cbf548bbf4
fa1f197ec3aeea94c630aa9950ac210aa30d58f4
